### PR TITLE
Login mechanism

### PR DESCRIPTION
**Pull request by BarbaraBochinska**
Done and done.

### CODE REVIEW

File: front/src/sideBard/LoggedOutSidebar.js [hunk 1]
```diff
@@ -21,40 +20,21 @@ export default class LoggedOutSidebar extends Component {
...
-              <Button variant="outline-secondary" block>Submit</Button>
-            </ButtonToolbar>
-            </FormGroup>
-            </Form>
-
-        </Dropdown.Item>
-          </Dropdown>
+              <Dropdown.Item eventKey="1-1" style={{ paddingLeft: 0 }} />
+              <LoginForm />
+            </Dropdown>
```
Comment (sznowak13) [inline front/src/sideBard/LoggedOutSidebar.js]: Ah, good, separating larger parts of components into smaller components.

File: front/src/sideBard/LoginForm.js [hunk 2]
```diff
@@ -32,12 +29,26 @@ export default class LoginForm extends Component {
...
     });
+    let status = response.status;
+    if (status === 400){
+      console.log("jeb sie");
+    } else if (status === 200){
+      let json = await response.json();
+      localStorage.setItem("muzi_muzi_token", json.token);
+    }
+
+      // .then(response => response.json())
```
Comment (BarbaraBochinska) [inline front/src/sideBard/LoginForm.js]: Do we need this? I suggest to remove unused code

File: front/src/sideBard/LoginForm.js [hunk 3]
```diff
@@ -32,12 +29,26 @@ export default class LoginForm extends Component {
     }
   }
 
-  authorizeUser() {
+  async authorizeUser() {
     let data = JSON.stringify(this.state);
-    fetch("http://localhost:8000/api-token/login", {
+    let response = await fetch("http://localhost:8000/api-token/login", {
       method: "POST",
```
Comment (sznowak13) [inline front/src/sideBard/LoginForm.js]: 🥇 async function <3

File: front/src/App.js [hunk 4]
```diff
@@ -18,9 +18,11 @@ class App extends Component {
...
           <Row>
             <Col md="auto">
-              {" "}
-              {/* <LoggedSidebar /> */}
-              <LoggedOutSidebar />
+              {localStorage.getItem("muzi_muzi_token") !== null ? (
+                <LoggedSidebar />
+              ) : (
+                <LoggedOutSidebar />
+              )}
```
Comment (sznowak13) [inline front/src/App.js]: Can we mayb separate this logic into method?

File: front/src/sideBard/LoginForm.js [hunk 5]
```diff
@@ -78,6 +82,7 @@ export default class LoginForm extends Component {
               variant="outline-secondary"
               block
               onClick={this.authorizeUser}
+              disabled={this.validateLoginInputs()}
             >
```
Comment (sznowak13) [inline front/src/sideBard/LoginForm.js]: I like this feature very much, good job!

File: front/src/sideBard/LoginForm.js (hunk 2 above)
Comment (sznowak13) [reply front/src/sideBard/LoginForm.js]: Yes, very good idea, I will push a commit with a fix.

Comment (sznowak13) [reply front/src/sideBard/LoginForm.js]: Um, I think that was from an older commit because its not present in the HEAD commit. Issue resolved.

**Pull request merged**